### ISSUE TITLE
OneDrive - recursive download of files from user's drive

### ISSUE THREAD

**Issue by ssmails** [vgrem/office365-rest-python-client#831]
Looking at the docs for Onedrive, it shows how we can download files from users root folder.
https://github.com/vgrem/Office365-REST-Python-Client

Is there a way to list and download all files from users root folder recursively without using any /me endpoints - can the examples be updated with it ? 

The examples shows a recursive version, but using the /me endpoint.
https://github.com/vgrem/Office365-REST-Python-Client/blob/master/examples/onedrive/folders/list_with_files.py
https://github.com/vgrem/Office365-REST-Python-Client/blob/master/examples/onedrive/folders/list_with_files.py#L16

@vgrem

**Comment by vgrem** [comment]
Yes - you can list and download another user's drive recursively without any /me endpoint. Resolve the drive via users/{upn}/drive, then walk and archive it with the same helpers as the /me examples:

```bash
pip install --upgrade office365-rest-python-client
```



```python
client = GraphClient(tenant=tenant) \
    .with_client_secret(client_id, client_secret) \
    .require_application_permission("Files.Read.All")

drive = client.users[user_upn].drive.get().execute_query()
root = drive.root.get().execute_query()

folders = root.get_folders(recursive=True).execute_query()
files = root.get_files(recursive=True).execute_query()   # pages automatically
print(len(files), "files")

with open("backup.zip", "wb") as f:
    root.download_folder(f).execute_query()
```

A runnable end-to-end example (inventory + zip download, no `/me`) is in the repo at [`examples/onedrive/drives/export_user_drive.py`](https://github.com/vgrem/office365-rest-python-client/blob/master/examples/onedrive/drives/export_user_drive.py).
